controller/src/main/java/org/jboss/as/controller/ValidateModelStepHandler.java

```diff
@@ -147,6 +150,62 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
     }
 
+    private void handleObjectAttributes(ModelNode model, AttributeDefinition attr, String absoluteParentName) throws OperationFailedException {
+        if (attr instanceof ObjectTypeAttributeDefinition) {
+            validateNestedAttributes(model.get(attr.getName()), attr, absoluteParentName);
+        } else if (attr instanceof ObjectListAttributeDefinition) {
+            AttributeDefinition valueType = ((ObjectListAttributeDefinition) attr).getValueType();
+            ModelNode list = model.get(attr.getName());
+            for (int i = 0; i < list.asInt(); i++) {
+                validateNestedAttributes(list.get(i), valueType, absoluteParentName + "[" + i + "]");
+            }
+        } else if (attr instanceof ObjectMapAttributeDefinition) {
+            AttributeDefinition valueType = ((ObjectMapAttributeDefinition) attr).getValueType();
+            ModelNode map = model.get(attr.getName());
```

Comment:
why not use 
Map<String, ModelNode>map = ((ObjectMapAttributeDefinition) attr).unwrap(...